File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let i = 0;
  let j = i+1;
  
  while (j !== undefined) {
      let sum = nums[i] + nums[j];
      if (sum === target) {
          return [i,j];
      } else if (nums[j]===undefined) {
          i++;
          j=i+1;
      } else {
          j++;
      }
  }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i in range(len(nums)):
            hash[nums[i]] = i
        for j in range(len(nums)):
            value = target-nums[j]
            if value in hash and hash[value] != j:
                return [j, hash[value]]